app/components/QRAddressScanner.jsx

```diff
@@ -0,0 +1,181 @@
+        this.setState({
+            visible: false
+        });
+    }
+
+    isBitcoinAddress(data) {
+        return /bitcoin:([a-zA-Z0-9]+)/.test(data);
+    }
+
+    parseBitcoinAddress(str) {
+        const address = str.match(/bitcoin:([a-zA-Z0-9]+)/);
+        const amount = str.match(/amount=([0-9\.]+)/);
+        return {
+            address: (address && address[1]) || null,
+            amount: (amount && amount[1]) || null
+        };
+    }
+
+    onScanSuccess(data) {
+        if (this.isBitcoinAddress(data)) {
+            let result = this.parseBitcoinAddress(data);
+            if (result) {
+                this.setState({
+                    address: result.address,
+                    amount: result.amount
+                });
+            }
+        } else {
+            this.setState({
+                address: data,
+                amount: null
+            });
+        }
+    }
+
+    retry() {
+        this.setState({
+            address: null,
+            amount: null
+        });
+    }
+
+    submit() {
+        this.handleClose();
+        if (typeof this.props.onSuccess === "function") {
+            this.props.onSuccess({
+                address: this.state.address,
+                amount: this.state.amount
+            });
+        }
+    }
+
+    render() {
+        const handleError = err => {
+            if (typeof this.props.onError === "function")
+                this.props.onError(err);
+        };
+
+        const handleScan = data => {
+            if (data) {
+                this.onScanSuccess(data);
+            }
+        };
+
+        return (
+            <div className="qr-address-scanner">
+                <button
+                    className="qr-address-scanner-btn"
+                    onClick={this.handleClick}
+                >
+                    <Icon name="photo-camera" />
+                </button>
+                {(this.state.visible && (
+                    <BaseModal
+                        className="qr-address-scanner-modal"
+                        modalHeader="global.scan_qr_code"
+                        id={this.modalId}
+                        overlay={true}
+                        noLoggo={true}
+                        onClose={this.handleClose}
+                    >
+                        <QrReader
+                            delay={100}
+                            onError={handleError}
+                            onScan={handleScan}
+                            style={{
+                                width: "calc(100% - 48px)",
+                                margin: "0 24px"
+                            }}
+                        />
+
+                        {this.state.address && (
+                            <div>
+                                <div className="qr-address-scanner-status">
+                                    <div className="qr-address-scanner-status-title">
+                                        Address found:
```

Comment:
You should add translation keys for all text entries. <Translate content="new_key" />